lms/djangoapps/bulk_email/docs/decisions/001-bulk-email-content-sanitization.rst

```diff
@@ -0,0 +1,31 @@
+====================================
+Bulk Email HTML Content Sanitization
+====================================
+
+Status
+------
+
+Accepted
+
+Background
+----------
+
+The Bulk Course Email tool allows course teams to author messages that can be sent to learners and supporting staff enrolled in their courses. An editor is provided as part of the tool for authoring these email messages. If desired, the editor allows advanced users to write custom raw HTML in the message.
+
+It is considered good security practice to scan and sanitize user-provided content before storing or using the data.
+
+Decision
+--------
+
+We will be sanitizing the HTML content sent through the bulk course email tool before sending the messages to any recipients.
+
+We will use the `bleach`_ Python package to sanitize the data.
+
+We will introduce a new configuration setting called ``BULK_COURSE_EMAIL_ALLOWED_HTML_TAGS`` that acts as an *allowlist* of HTML tags permitted for use within the body of a message authored by the bulk course email tool. A default list of options is provided in the ``lms/envs/common.py`` configuration file. Offending data will be escaped (converted to plaintext) over being stripped out.
```

Comment:
Can we add a link for lms/envs/common.py with a line number? It's a huge file.

Reply:
Sure, I can do that!